jujugui/static/gui/src/test/test_app.js

```diff
@@ -1386,6 +1386,30 @@ describe('App', function() {
       app.destroy();
     });
 
+    it('can pick the right environment from a list based on config',
```

Comment:
I feel like there is at least one missing test here because there is an untested branch in the conditionals.